### PR TITLE
Added pronunciation

### PR DESCRIPTION
**Pull request by VIEWVIEWVIEW**
Certain words have a pronunciation attribute, e.g. [keyboard](https://www.duden.de/rechtschreibung/Keyboard), where you can find the pronunciation in [IPA notation](https://en.wikipedia.org/wiki/International_Phonetic_Alphabet).

HTML looks like this:
```html
<span class="ipa">[ˈkiːbɔːɐ̯t]</span>
```

Example:
```
$ python3 run_duden.py Keyboard
Keyboard, das
=============
Pronunciation: [ˈkiːbɔːɐ̯t]
Word type: Substantiv, Neutrum
Commonness: 2/5
Separation: Key|board
Meaning overview:
 0. elektronisch verstärktes Tasteninstrument

 1. [Geräteteil mit der] Tastatur (b) eines Personal Computers```

### CODE REVIEW

**Comment by radomirbosak** [comment]
Thank you for this pull request! This is a great addition. :)

**Comment by radomirbosak** [comment]
Could you also add a CLI option `--phonetic` so that this new property is usable from the CLI like this?
```shell
$ ./run_duden.py --phonetic Petersilie
[petɐˈziːli̯ə]
```

**Comment by radomirbosak** [comment]
Also, could you update the tests so that this new feature is resilient to change?

In this case, first the new property has to be added to `test_online_attributes.py` with other `basic_attributes` and the expected value should be updated in all `tests/test_data/*.yaml`.

From my testing, apart from Keyboard, the words Petersilie, Qat and einfach from the test data had a non-trivial pronunciation data.

Also, maybe it would be useful to add a new test data point, the word keyboard. Test data can be easily generated with the `--export` option like this:
```
./run_duden.py --export Keyboard > tests/test_data/Keyboard.yaml
```

File: duden/display.py [hunk 1]
```diff
@@ -122,6 +122,8 @@ def describe_word(word):
     print(yellow(word.title, bold=True))
     print(yellow('=' * len(word.title)))
 
+    if word.phonetic:
+         print(white(_('Pronunciation:'), bold=True), word.phonetic)
```
Comment (radomirbosak) [inline duden/display.py:126]: autopep8 reports an extra space here:
```diff
$ make check
autopep8 --max-line-length 99  --diff -r duden/ tests/ | colordiff
--- original/duden/display.py
+++ fixed/duden/display.py
@@ -123,7 +123,7 @@
     print(yellow('=' * len(word.title)))
 
     if word.phonetic:
-         print(white(_('Pronunciation:'), bold=True), word.phonetic)
+        print(white(_('Pronunciation:'), bold=True), word.phonetic)
     if word.part_of_speech:
         print(white(_('Word type:'), bold=True), word.part_of_speech)
     if word.usage:
...
```

File: duden/word.py [hunk 2]
```diff
@@ -344,3 +344,14 @@ def words_before(self):
     def words_after(self):
         """Returns 5 words after this one in duden database"""
         return [name for name, _ in self.before_after_structure['Im Alphabet danach']]
+
+    @property
+    def phonetic(self):
+        """
+        Returns pronunciation of the word in phonetic notation (https://en.wikipedia.org/wiki/International_Phonetic_Alphabet)
```
Comment (radomirbosak) [inline duden/word.py:351]: Could you please wrap this line to maximum 99 characters? `flake8` checker complains:
```shell
$ make check
autopep8 --max-line-length 99  --diff -r duden/ tests/ | colordiff
flake8 --builtins="_" --max-line-length 99 duden/ tests/
duden/word.py:351:100: E501 line too long (126 > 99 characters)
make: *** [Makefile:11: check] Error 1
```

**Review by radomirbosak — CHANGES_REQUESTED** [review]

**Comment by VIEWVIEWVIEW** [comment]
I'll check the tests on Sunday, for some reason a lot of them are failing for me.

**Comment by radomirbosak** [comment]
Sorry, the failing tests problem is probably not faul, I can reproduce it on `master` branch too.

```shell
$ make test
...
====================================================== short test summary info =======================================================
FAILED tests/test_online_attributes.py::test_basic_attributes[title-None-expected_dict7] - AttributeError: 'NoneType' object has no...
FAILED tests/test_online_attributes.py::test_basic_attributes[name-None-expected_dict7] - AttributeError: 'NoneType' object has no ...
FAILED tests/test_online_attributes.py::test_basic_attributes[article-None-expected_dict7] - AttributeError: 'NoneType' object has ...
FAILED tests/test_online_attributes.py::test_basic_attributes[part_of_speech-None-expected_dict7] - AttributeError: 'NoneType' obje...
FAILED tests/test_online_attributes.py::test_basic_attributes[frequency-None-expected_dict7] - AttributeError: 'NoneType' object ha...
FAILED tests/test_online_attributes.py::test_basic_attributes[usage-None-expected_dict7] - AttributeError: 'NoneType' object has no...
FAILED tests/test_online_attributes.py::test_basic_attributes[word_separation-None-expected_dict7] - AttributeError: 'NoneType' obj...
FAILED tests/test_online_attributes.py::test_basic_attributes[synonyms-None-expected_dict7] - AttributeError: 'NoneType' object has...
FAILED tests/test_online_attributes.py::test_basic_attributes[origin-None-expected_dict7] - AttributeError: 'NoneType' object has n...
FAILED tests/test_online_attributes.py::test_basic_attributes[words_before-parsed_word1-expected_dict1] - AssertionError: assert ['...
FAILED tests/test_online_attributes.py::test_basic_attributes[words_before-None-expected_dict7] - AttributeError: 'NoneType' object...
FAILED tests/test_online_attributes.py::test_basic_attributes[words_after-None-expected_dict7] - AttributeError: 'NoneType' object ...
FAILED tests/test_online_attributes.py::test_meaning_overview[None-expected_dict7] - AttributeError: 'NoneType' object has no attri...
FAILED tests/test_online_attributes.py::test_word_compounds[None-expected_dict7] - AttributeError: 'NoneType' object has no attribu...
FAILED tests/test_online_attributes.py::test_word_grammar[parsed_word6-expected_dict6] - AssertionError: assert [({'Indikativ.../di...
FAILED tests/test_online_attributes.py::test_word_grammar[None-expected_dict7] - AttributeError: 'NoneType' object has no attribute...
=================================================== 16 failed, 97 passed in 29.38s ===================================================
make: *** [Makefile:7: test] Error 1
```
let me try to fix it during Saturday. Either the word data or page layout changed recently, so the test started failing.
I created an issue for fixing the tests here: https://github.com/radomirbosak/duden/issues/138

File: duden/cli.py [hunk 3]
```diff
@@ -114,6 +117,8 @@ def parse_args():
 
     parser.add_argument('-V', '--version', action='store_true',
                         help=_('print program version'))
+    parser.add_argument('--phonetic', action='store_true',
+                        help=('displays pronunciation'))
```
Comment (radomirbosak) [inline duden/cli.py:121]: In this codebase, `_` is a function that localizes strings to various languages.

```suggestion
                        help=_('display pronunciation'))
```

**Review by radomirbosak — CHANGES_REQUESTED** [review]
Thank you for making the CLI changes.

For the failing tests, I fixed them earlier today in master, so if you pull in the changes from current master, the tests should pass.

I also added a Makefile goal [update-test-data](https://github.com/radomirbosak/duden/blob/master/Makefile#L40-L47), so I think adding the `Keyboard` test case should be as simple as adding `Keyboard` to the `update-test-data` and running:
```shell
$ make update-test-data
```

Please let me know if there are any unexpected errors.

File: duden/cli.py (hunk 3 above)
Comment (radomirbosak) [reply duden/cli.py:121]: (trivial)
Btw. if you know german, spanish or esperanto, please translate the strings
* `'Pronunciation:'`
* `'display pronunciation'`
in the `duden/locale/*.po` files

If you don't, no problem, the localization will default to the default string (english) if not supplied.

File: duden/cli.py [hunk 4]
```diff
@@ -114,6 +117,8 @@ def parse_args():
 
     parser.add_argument('-V', '--version', action='store_true',
                         help=_('print program version'))
+    parser.add_argument('--phonetic', action='store_true',
```
Comment (radomirbosak) [inline duden/cli.py:120]: One more thing, could you please alse add the `--phonetic` switch to shell completion scripts?
https://github.com/radomirbosak/duden/blob/master/completions/duden
https://github.com/radomirbosak/duden/blob/master/completions/duden.fish

**Comment by VIEWVIEWVIEW** [comment]
Everything should be done except localization. If I missed something, let me know.

How exactly does the numbering work in the `duden.pot` and `de_DE.po` files work?

For example, here it says `main.py:79`, although neither the file where the string is located is called `main.py` nor is it line 79:
https://github.com/radomirbosak/duden/blob/ddbdaa8037c95de6b44245d459b42c619b001990/duden/locale/duden.pot#L18-L20

actual source:
https://github.com/radomirbosak/duden/blob/ddbdaa8037c95de6b44245d459b42c619b001990/duden/display.py#L124-L126

**Comment by radomirbosak** [comment]
Thank you very much for the changes! I tested it and everything looks good!

For the localization part, I believe the comments like `main.py:79` are ignored during execution. They were generated by some tool during the first time localization was added to this project, and the string positions changed since. I think we can ignore those lines, but if you want, you can specify the current string location for the new string.

Btw. testing localization could work like this:
1. add to duden.pot
```
msgid "display pronunciation"
msgstr ""
```
2. add to *.po
```
msgid "display pronunciation"
msgstr "[Uebersetzung]"
```
3. compile translations
```shell
$ make localization
```
4. test with `run_duden.py`
```shell
$ ./run_duden.py --help
$ LANG=de_DE.UTF-8 ./run_duden.py --help
$ LANG=es_ES.UTF-8 ./run_duden.py --help
$ LANG=eo_EO.UTF-8 ./run_duden.py --help
```
(I should document this in the README or somewhere)

**Review by radomirbosak — APPROVED** [review]

**Comment by radomirbosak** [comment]
In any case, I'm fine with merging this PR as it is. So please let me know if you want to add the translations or if I can merge it now.

**Comment by VIEWVIEWVIEW** [comment]
> 
> 
> In any case, I'm fine with merging this PR as it is. So please let me know if you want to add the translations or if I can merge it now.

Merge it :)

**Comment by radomirbosak** [comment]
Merged and released to pypi.

To upgrade the package, run:
```shell
$ pip3 install --user --upgrade duden
```

Thank you very much @VIEWVIEWVIEW for this contribution :)

**Pull request merged**